web/src/app/services/IntegrationKeyForm.tsx

```diff
@@ -13,11 +12,29 @@ import { AppLink } from '../util/AppLink'
 
 const useStyles = makeStyles((theme) => ({
   infoIcon: {
-    color: theme.palette.primary['500'],
+    color: theme.palette.primary.main,
   },
 }))
 
-export default function IntegrationKeyForm(props) {
+interface Value {
+  name: string
+  type: ['generic' | 'grafana' | 'site24x7' | 'email']
```

Comment:
Since these are optional in `propTypes`, we should make them optional here via `name?:` and `type?:`

Reply:
I think they were meant to be required but got missed.

For `type` I think it needs to be without the square brackets since it shouldn't be an array